ckanext/saml2/plugin.py

```diff
@@ -353,15 +404,22 @@ def _create_or_update_user(self, user_name, saml_info):
         context = {'schema': user_schema, 'ignore_auth': True}
 
         if is_new_user:
+            username = _get_random_username_from_email(saml_info['email'][0])
+            gen = saml_info['id'][0]
+            data_dict['name'] = username
```

Comment:
I've asked you before. On L376 username is used to obtain user from db. If you are generating the save value - why don't you use same variable as well.
If it different values - L376 unable to finde you new user in feature, right? and you end up creating infinite amount of users